homeassistant/components/modbus/binary_sensor.py

```diff
@@ -61,10 +111,46 @@ async def async_update(self, now: datetime | None = None) -> None:
                 return
             self._lazy_errors = self._lazy_error_count
             self._attr_available = False
-            self.async_write_ha_state()
-            return
+            self._result_bits = None
+        else:
+            self._lazy_errors = self._lazy_error_count
+            self._attr_is_on = result.bits[0] & 1
+            self._attr_available = True
+            self._result_bits = result.bits
 
-        self._lazy_errors = self._lazy_error_count
-        self._attr_is_on = result.bits[0] & 1
-        self._attr_available = True
         self.async_write_ha_state()
+        if self._coordinator:
+            await self._coordinator.async_refresh()
+
+
+class SlaveSensor(CoordinatorEntity, RestoreEntity, BinarySensorEntity):
+    """Modbus slave binary sensor."""
+
+    def __init__(
+        self, coordinator: DataUpdateCoordinator[Any], idx: int, entry: dict[str, Any]
+    ) -> None:
+        """Initialize the Modbus binary sensor."""
+        idx += 1
+        self._attr_name = f"{entry[CONF_NAME]}_{idx}"
```

Comment:
Since it's a name, maybe drop the `_` in favor of a `<space>` ? 
```suggestion
        self._attr_name = f"{entry[CONF_NAME]} {idx}"
```

Reply:
I would love to do that, but when I try calling hass.states.get("binary_sensor.mySensor 1").state, it breaks down, it seems it does not allow spaces.

Reply:
Entity ID is derived from the name when it's added to the state machine. It will automatically add the `_` in the entity ID, but the UI will show it with a space.